src/track/globaltrackcache.cpp

```diff
@@ -576,98 +630,118 @@ TrackRef GlobalTrackCache::initTrackId(
     return trackRefWithId;
 }
 
-void GlobalTrackCache::evictOrDelete(
+void GlobalTrackCache::evictAndSave(
         Track* plainPtr) {
     DEBUG_ASSERT(plainPtr);
+
+    // Nearly everything is possible until the cache is locked!!!
     GlobalTrackCacheLocker cacheLocker;
 
-    const AllocatedTracks::iterator allocatedTrack =
-            m_allocatedTracks.find(plainPtr);
-    if (allocatedTrack == m_allocatedTracks.end()) {
-        // We have handed out and already delete this track while waiting at
+    // After obtaining the lock we need to check if the deletion
+    // is still valid, i.e. if the pointer has not already been
+    // deleted or if it has been revived by an intermediate cache
+    // lookup while this thread was blocked.
+    const CachedTracks::iterator cachedTrack =
+            m_cachedTracks.find(plainPtr);
+    if (cachedTrack == m_cachedTracks.end()) {
+        // We have already deleted this track while waiting at
```

Comment:
// We have already evicted this track ...

The second smart pointer for deleting may still live.